server/controllers/student.py

```diff
@@ -100,15 +100,29 @@ def code(cid, aid, bid):
         backup = Backup.query.get(bid)
         if backup and backup.can_view(current_user, group, assgn):
             submitter = User.query.get(backup.submitter)
-            file_contents = [m for m in backup.messages if
-                                m.kind == "file_contents"]
-            if file_contents:
-                files = file_contents[0].contents
-                return render_template('student/assignment/code.html', course=course,
-                        assignment=assgn, backup=backup, submitter=submitter,
-                        files=files)
-            else:
-                flash("That code submission doesn't contain any code")
+
+            diff = Diff.query.filter_by(assignment=aid, backup=bid).one_or_none()
```

Comment:
We should only generate Diff objects for assignments that have templates. This might mean that we have a separate template just for diffs.